pkg/webhook/v1beta1/experiment/validator/validator.go

```diff
@@ -67,6 +67,14 @@ func (g *DefaultValidator) InjectClient(c client.Client) {
 // ValidateExperiment validates experiment for the given instance.
 // oldInst is specified when experiment is edited.
 func (g *DefaultValidator) ValidateExperiment(instance, oldInst *experimentsv1beta1.Experiment) error {
+	namingConvention, _ := regexp.Compile("^[a-z]([-a-z0-9]*[a-z0-9])$")
```

Comment:
Should we follow the regex syntax from the controller-runtime suggestion ?
```suggestion
	namingConvention, _ := regexp.Compile("[a-z]([-a-z0-9]*[a-z0-9])?")
```

Reply:
I'm not an expert with golang, I'm a bit confused with this:
```
package main

import (
	"fmt"
	"regexp"
)

func main() {
	namingConvention, _ := regexp.Compile("[a-z]([-a-z0-9]*[a-z0-9])?")
	fmt.Println(namingConvention.MatchString("test")) // true
	fmt.Println(namingConvention.MatchString("test-1234")) // true
	fmt.Println(namingConvention.MatchString("1234-1234")) // false
	fmt.Println(namingConvention.MatchString("1234-test")) // true!! this is what i didn't expected
	
	nc2, _ := regexp.Compile("^[a-z]([-a-z0-9]*[a-z0-9])?") // '^' was added
	fmt.Println(nc2.MatchString("test")) // true
	fmt.Println(nc2.MatchString("test-1234")) // true
	fmt.Println(nc2.MatchString("1234-1234")) // false
	fmt.Println(nc2.MatchString("1234-test")) // false!!, so I followed this regexp pattern.

}
```

That's why I used the pattern `"^[a-z]([-a-z0-9]*[a-z0-9])?"`, not `"[a-z]([-a-z0-9]*[a-z0-9])?"`

Reply:
@anencore94 Thank you for investigating this, you are right.
As I can see, kubebuilder also adds `^` and `$` to the regex: https://github.com/kubernetes-sigs/kubebuilder/blob/master/pkg/internal/validation/dns.go#L45.

Reply:
@anencore94 do you want to keep your first regex: `^[a-z]([-a-z0-9]*[a-z0-9])$` or with `?` at the end ?

Reply:
I think my first regex was wrong since this case:
```
package main

import (
	"fmt"
	"regexp"
)

func main() {
	nc1, _ := regexp.Compile("^[a-z]([-a-z0-9]*[a-z0-9])?")
	fmt.Println(nc1.MatchString("test")) // true
	fmt.Println(nc1.MatchString("test-1234")) // true
	fmt.Println(nc1.MatchString("1234-1234")) // false
	fmt.Println(nc1.MatchString("1234-test")) // false
	fmt.Println(nc1.MatchString("t")) // true, That's what we expected
		
	fmt.Println("@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@")
	
	nc2, _ := regexp.Compile("^[a-z]([-a-z0-9]*[a-z0-9])$")
	fmt.Println(nc2.MatchString("test")) // true
	fmt.Println(nc2.MatchString("test-1234")) // true
	fmt.Println(nc2.MatchString("1234-1234")) // false
	fmt.Println(nc2.MatchString("1234-test")) // false
	fmt.Println(nc2.MatchString("t")) // false!! this is what we didn't expected

	fmt.Println("@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@")
	
	nc3, _ := regexp.Compile("^[a-z]([-a-z0-9]*[a-z0-9])?$")
	fmt.Println(nc3.MatchString("test")) // true
	fmt.Println(nc3.MatchString("test-1234")) // true
	fmt.Println(nc3.MatchString("1234-1234")) // false
	fmt.Println(nc3.MatchString("1234-test")) // false
	fmt.Println(nc3.MatchString("t")) // true, That's what we expected
}
```

I'm not sure what to use between `nc` and `nc3`.

But I guess we should use `nc3` : `^[a-z]([-a-z0-9]*[a-z0-9])?$`, since you mentioned, It seems golang always needs `^` and `$`.